tests/prague/eip6206_jumpf/helpers.py

```diff
@@ -0,0 +1,67 @@
+"""
+EOF V1 Code Validation tests
+"""
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    EOFTestFiller,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+)
+from ethereum_test_tools.eof.v1 import Container
+
+
+def execute_tests(state_test: StateTestFiller, eof_test: EOFTestFiller, container: Container):
+    """
+    Executes EOF validaiton on the container, and if valid executes the container, checking for
+    a canary of {0, 1} for successful execution.
+    """
+    eof_test(
+        data=container,
+        expect_exception=container.validity_error,
+    )
+
+    if container.validity_error is None:
+        env = Environment()
+
+        pre = {
+            TestAddress: Account(
+                balance=1000000000000000000000,
+                nonce=1,
+            ),
+            Address(0x100): Account(
+                code=container,
+                nonce=1,
+            ),
+        }
+
+        tx = Transaction(
+            nonce=1,
+            to=Address(0x100),
+            gas_limit=44_000,
+            gas_price=10,
+            protected=False,
+            data="1",
+        )
+
+        post = {Address(0x100): Account(storage={0: 1})}
+
+        state_test(
```

Comment:
ah, interesting. @marioevz  there are plans to auto extend eof_test  to generate state and blockchain tests automatically

Reply:
Yes, I think this logic could be embedded in `eof_test`.
We could embed two steps of execution in a transaction:
1) Attempting to deploy the EOF container, and if:
  - valid: verify the container was deployed
  - invalid: verify the container was not deployed
2) Execute the deployed code:
  - We could also add a `post` field to the `eof_test` so each test function can define what to expect after the container is executed.

This could all be added into src/ethereum_test_tools/spec/eof/eof_test.py

Reply:
Added `EOFStateTest`.